client/task-list/tasks/PaymentGatewaySuggestions/index.js

```diff
@@ -82,6 +82,16 @@ export const PaymentGatewaySuggestions = ( { query } ) => {
 		}
 	);
 
+	useEffect( () => {
+		if ( paymentGateways.size ) {
+			recordEvent( 'tasklist_payments_options', {
```

Comment:
This event fires on every single component update, which is quite a lot.  I think we'll need to memoize `mappedSuggestions` and possibly extract it out of `useSelect` to prevent this constant firing.

<img width="721" alt="Screen Shot 2021-07-15 at 10 28 40 AM" src="https://user-images.githubusercontent.com/10561050/125805904-222dbf7e-43b6-4ce8-8267-4a3049e9377b.png">

Reply:
Good call @joshuatf , made this change in https://github.com/woocommerce/woocommerce-admin/pull/7304/commits/3181c7cec9dbb0399c4d0eb918f545b7510feff5.